File: implementations/java/src/MySQLBulk.java

import java.sql.*;

public class MySQLBulk {

    public static void main(String[] args) {
        StringBuilder sb = new StringBuilder("INSERT INTO tb_java_bulk(hash) VALUES ");
        String query;
        int i, hash;
        try (
            Connection conn = DriverManager.getConnection(
                    "jdbc:mysql://127.0.0.1:3306/hashes?allowMultiQueries=true",
                    "root",
                    "root123"
            );
            Statement stmt = conn.createStatement();    
        ) {
            stmt.executeUpdate("DROP TABLE IF EXISTS tb_java_bulk;"
                    + "CREATE TABLE tb_java_bulk"
                    + "(id INTEGER AUTO_INCREMENT PRIMARY KEY,"
                    + " hash INTEGER NOT NULL)");

            conn.setAutoCommit(false);

            for (i = 1; i <= 10000; i++) {
                hash = Hash.hash32(i);
                sb.append(String.format("(%d),", hash));
            }

            query = sb.toString().replaceAll(".$", ";");

            stmt.execute(query);

            conn.commit();

        } catch (Exception e) {
            System.err.println(e.getMessage());
        }
    }
}

File: implementations/py/src/mysql-bulk.py
from hashes import hash32
from mysql.connector import connect

try:
    with connect(
        host="127.0.0.1",
        user="root",
        password="root123",
        database="hashes",
    ) as conn:
        with conn.cursor() as cur:
            cur.execute(
                """DROP TABLE IF EXISTS tb_py_bulk;
                CREATE TABLE tb_py_bulk
                (id INTEGER AUTO_INCREMENT PRIMARY KEY,
                hash INTEGER NOT NULL);"""
            )
            conn.next_result()
            conn.commit()

            buf = []
            for i in range(1, 10_001):
                hash = hash32(i)
                buf.append(f"({hash})")

            query = (
                "BEGIN;\nINSERT INTO tb_py_bulk(hash) VALUES "
                + ",".join(buf)
                + ";\nCOMMIT;"
            )
            cur.execute(query)

except Exception as e:
    print(e)
